net/api/impl/posix.h

```diff
@@ -86,8 +88,8 @@ class HTTPClientPOSIX final {
       // Don't uncomment the next line!
       // connection.SendEOF();
       message_.reset(new HTTPRedirectableReceivedMessage(connection));
-      // TODO(dkorolev): Rename `URL()`, it's only called so now because of HTTP request/response format.
-      const int response_code_as_int = atoi(message_->URL().c_str());
+      // TODO(dkorolev): Rename `Path()`, it's only called so now because of HTTP request/response format.
+      const int response_code_as_int = atoi(message_->Path().c_str());
       response_code_ = static_cast<HTTPResponseCode>(response_code_as_int);
       if (response_code_as_int >= 300 && response_code_as_int <= 399 && !message_->location.empty()) {
```

Comment:
It's parsed in ```OnHeader()``` of ```struct HTTPRedirectHelper :
HTTPDefaultHelper```.
ᐧ

Thanks,
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783


On Wed, Jan 7, 2015 at 9:40 AM, John Babak <notifications@github.com> wrote:

> In net/api/impl/posix.h
> <https://github.com/KnowSheet/Bricks/pull/36#discussion-diff-22609874>:
>
> > @@ -86,8 +88,8 @@ class HTTPClientPOSIX final {
> >        // Don't uncomment the next line!
> >        // connection.SendEOF();
> >        message_.reset(new HTTPRedirectableReceivedMessage(connection));
> > -      // TODO(dkorolev): Rename `URL()`, it's only called so now because of HTTP request/response format.
> > -      const int response_code_as_int = atoi(message_->URL().c_str());
> > +      // TODO(dkorolev): Rename `Path()`, it's only called so now because of HTTP request/response format.
> > +      const int response_code_as_int = atoi(message_->Path().c_str());
> >        response_code_ = static_cast<HTTPResponseCode>(response_code_as_int);
> >        if (response_code_as_int >= 300 && response_code_as_int <= 399 && !message_->location.empty()) {
>
> The value of the Location header would more logically be a part of some
> headers hashmap, not a property of the message (though it makes things
> more complex).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KnowSheet/Bricks/pull/36/files#r22609874>.
>